doc/lightning-sendpay.7.md

```diff
@@ -27,8 +27,8 @@ definite failure.
 The *label* and *bolt11* parameters, if provided, will be returned in
 *waitsendpay* and *listsendpays* results.
 
-The *msatoshi* amount, if provided, is the amount that will be recorded
-as the target payment value. If not specified, it will be the final
+The *msatoshi* amount must be provided if *partid* is non-zero, otherwise
```

Comment:
This parameter used to be used for value randomization, where we overpay the payee, but record the exact amount indicated in the invoice in our database rather than the overpaid amount (i.e. treating the overpayment as a fee).  We have since replaced value randomization with shadow routing, also overpaying the payee, but it seems this was no longer used to override the database record.  I think it is better if our permanent records did not reveal the use of overpayment to obscure the location of the payee, but well.  In any case changing `msatoshi` to make it signal multipath is a breaking change in the previous interface.

Reply:
Yes, the original intent with mpp was to only override the semantic for msatoshi for this case, but since it's broken in 0.8.0 anyway :(

The original use has been largely superceded by the fact that we now save the bolt11 string with the payment.